dipy/reconst/ivim.py

```diff
@@ -0,0 +1,391 @@
+            e_s += "bounds. Please update to Scipy 0.17 to use bounds"
+            raise ValueError(e_s)
+        else:
+            self.bounds = (np.array([0., 0., 0., 0.]),
+                           np.array([np.inf, 1., 0.1, 0.1]))
+
+    @multi_voxel_fit
+    def fit(self, data, mask=None):
+        """ Fit method of the Ivim model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+
+        Returns
+        -------
+        IvimFit object
+
+        References
+        ----------
+        .. [1] Federau, Christian, et al. "Quantitative measurement
+                   of brain perfusion with intravoxel incoherent motion
+                   MR imaging." Radiology 265.3 (2012): 874-881.
+        """
+        # Call the function _estimate_S0_D to get initial x0 guess.
+        x0 = self.estimate_x0(data)
+        # Use leastsq to get ivim_params
+        params_in_mask = self._leastsq(data, x0)
+        return IvimFit(self, params_in_mask)
+
+    def predict(self, ivim_params):
+        """
+        Predict a signal for this IvimModel class instance given parameters.
+
+        Parameters
+        ----------
+        ivim_params : array
+            The ivim parameters as an array [S0, f, D_star and D]
+
+        Returns
+        -------
+        ivim_signal : array
+            The predicted IVIM signal using given parameters.
+        """
+        return ivim_function(ivim_params, self.gtab.bvals)
+
+    def estimate_x0(self, data):
+        """
+        Fit the ivim params using a two stage fit.
+
+        In the two stage fitting routine, initially, we fit the signal
+        at bvals less than the specified split_b using the TensorModel
+        and get an intial guess for f and D. Then, using these parameters
+        we fit the entire data for all bvalues. The default split_b is
+        200.
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        Returns
+        -------
+        x0_guess : array
+            An array with initial values of S0, f, D_star, D for each voxel.
+        """
+        S0_hat, D_guess = self._estimate_S0_D(data)
+        f_guess = 1 - S0_hat / data[0]
+        # We set the S0 guess as the signal value at b=0
+        # The D* guess is roughly 10 times the D value
+        x0 = np.array([data[0], f_guess, 10 * D_guess, D_guess])
+
+        if self.bounds is None:
+            bounds_check = [np.inf, 1., 0.1, 0.1]
```

Comment:
This line is not currently being run in the tests. Please write a test that will check this condition.